src/core/VectorGraphModel.cpp

```diff
@@ -0,0 +1,2411 @@
+	return m_isSaveable;
+}
+bool VectorGraphDataArray::getIsNonNegative()
+{
+	return m_isNonNegative;
+}
+QColor* VectorGraphDataArray::getLineColor()
+{
+	return &m_lineColor;
+}
+QColor* VectorGraphDataArray::getActiveColor()
+{
+	return &m_activeColor;
+}
+QColor* VectorGraphDataArray::getFillColor()
+{
+	return &m_fillColor;
+}
+QColor* VectorGraphDataArray::getAutomatedColor()
+{
+	return &m_automatedColor;
+}
+int VectorGraphDataArray::getEffectorArrayLocation()
+{
+	return m_effectorLocation;
+}
+bool VectorGraphDataArray::getIsAnEffector()
+{
+	return m_isAnEffector;
+}
+int VectorGraphDataArray::getId()
+{
+	return m_id;
+}
+
+
+// array:
+
+int VectorGraphDataArray::add(float newX)
+{
+	int location = -1;
+	if (m_isFixedSize == true || m_dataArray.size() >= m_parent->getMaxLength()) { return location; }
+
+	bool found = false;
+	bool isBefore = false;
+	location = getNearestLocation(newX, &found, &isBefore);
+	if (found == false)
+	{
+		int targetLocation = -1;
+		bool dataChangedVal = false;
+		// if getNearestLocation returned a value
+		if (location >= 0)
+		{
+			targetLocation = location;
+			// shift the new data if the closest data x is bigger
+			// (done for swaping)
+			if (isBefore == true)
+			{
+				// we are adding one value, so dataArray.size() will be a valid location
+				if (targetLocation < static_cast<int>(m_dataArray.size()))
+				{
+					targetLocation++;
+				}
+			}
+			m_dataArray.push_back(VectorGraphPoint(newX, 0.0f));
+			swap(m_dataArray.size() - 1, targetLocation, true);
+			dataChangedVal = true;
+		}
+		else if (m_dataArray.size() <= 0)
+		{
+			m_dataArray.push_back(VectorGraphPoint(newX, 0.0f));
+			targetLocation = 0;
+			dataChangedVal = true;
+		}
+		location = targetLocation;
+
+		if (m_dataArray.size() <= 2)
+		{
+			formatDataArrayEndPoints();
+			dataChangedVal = true;
+		}
+		if (dataChangedVal == true)
+		{
+			// addtition breaks the order of the locations
+			// in m_needsUpdating, so we update the whole m_dataArray
+			getUpdatingFromPoint(-1);
+			dataChanged();
+		}
+	}
+	return location;
+}
+
+void VectorGraphDataArray::deletePoint(unsigned int pointLocation)
+{
+	if (m_isFixedSize == true || pointLocation >= m_dataArray.size()) { return; }
+
+	// deleting the points automationModel
+	deleteAutomationModel(m_dataArray[pointLocation].m_automationModel, true);
+	// swaping the point to the last location
+	// in m_dataArray
+	swap(pointLocation, m_dataArray.size() - 1, true);
+	m_dataArray.pop_back();
+	if (pointLocation == 0 || pointLocation == m_dataArray.size())
+	{
+		formatDataArrayEndPoints();
+	}
+	// calling clearedEvent
+	if (m_dataArray.size() == 0)
+	{
+		clearedEvent();
+	}
+	// deletion breaks the order of the locations
+	// in m_needsUpdating, so we update the whole m_dataArray
+	getUpdatingFromPoint(-1);
+	dataChanged();
+}
+
+void VectorGraphDataArray::formatArray(std::vector<PointF>* dataArrayOut, bool shouldClamp, bool shouldRescale, bool shouldSort, bool callDataChanged)
```

Comment:
you can break this function down into sortArray, RescaleArray, clampArray, and call these helpers from within this function, will get rid of codefactor warning.